### PR TITLE
linux-yocto-onl/5.15: update 5.15.35

### PR DESCRIPTION
**Pull request by KanjiMonster**
Update linux to 5.15.35 and meta to the appropriate revision.

Unfortunately yocto kernel-meta upstream enabled various subsystem
automatically when enabling virtio, some of which we don't need,
like DRM - there are no GPUs to pass through on a headless system.

Lacking a way to patch the file, add a pre-change version of virtio.cfg
so we can keep DRM disabled.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

### CODE REVIEW

**Comment by KanjiMonster** [comment]
Boot tested on as4610 and as4630.

**Comment by ideaship** [comment]
According to the commit message for 917043019b46d8f1d0a9808fe3b0b815f14e9e60, the patch added prerequisites for the virtio  drivers. So what are the consequences of reverting it? No virtio?

**Comment by KanjiMonster** [comment]
> According to the commit message for 917043019b46d8f1d0a9808fe3b0b815f14e9e60, the patch added prerequisites for the virtio drivers. So what are the consequences of reverting it? No virtio?

No, the prerequisites for (some) virtio drivers aren't automatically enabled, so e.g. before if you e.g. wanted PCI virtio, you needed to have it enabled in your base config, with this commit everything is automatically enabled, whether you want it or not.

In our case it forces DRM (= video) support on, which doesn't make sense for us because we don't have any GPUs to virtualize (at least on the ARM platforms).

**Comment by ideaship** [comment]
I am wondering how we maintain this as it evolves. Could it be easier to disable virtio.scc (or push an empty virtio.cfg) and then add our own virtio config files when we need them? Looks like we'll have to do that anyway. Or are our use cases covered by the old virtio.cfg that this commit now uses to override the new one?

**Comment by KanjiMonster** [comment]
It keeps the status-quo from before the changes. The ideal solution is probably to prod upstream to split the virtio into multiple smaller snippets (so you can selectively enable the subsystems you care about).

Currently virtio gets enabled included as a side effect of enabling the kernel feature virtualization, so we don't have a direct control over it.

**Comment by ideaship** [comment]
From an outcome-based perspective, yes, we are preserving the status quo. But we are effectively forking a file that is maintained upstream, and I don't see _how_ we are going to maintain it (i.e., according to which rules). If we freeze it the way it is now, that seems quite arbitrary. But if we are ever going to change it, what are the rules? When would we add or remove something?

**Comment by KanjiMonster** [comment]
> From an outcome-based perspective, yes, we are preserving the status quo. But we are effectively forking a file that is maintained upstream, and I don't see _how_ we are going to maintain it (i.e., according to which rules). If we freeze it the way it is now, that seems quite arbitrary. But if we are ever going to change it, what are the rules? When would we add or remove something?

There are no rules. This is (hopefully) only a temporary stop gap solution until we have a better one/worked with upstream to fix this.

**Comment by KanjiMonster** [comment]
What explanation? I really am against trying to "document" yocto stuff in our recipes, and litter docu there.

File: recipes-kernel/linux/linux-yocto-onl_5.15.bb [hunk 1]
```diff
@@ -4,16 +4,25 @@ require linux-yocto-onl.inc
...
     file://bisdn-kmeta;type=kmeta;name=bisdn-kmeta;destsuffix=bisdn-kmeta \
 "
 
+# "Revert" 917043019b46 ("virtio: Add prereqs for tiny") from yocto-kernel-cache
+# as it forces several subsystems like DRM on, which we do not need since none
+# of our devices have a GPU.
+# We cannot use patches for that since patches are applied after parsing meta,
+# so the next best thing is copying a pre-commit version of the file.
+SRC_URI += "\
+    file://kernel-meta;type=kmeta;name=kernel-meta;destsuffix=kernel-meta \
```
[the remark this answers is not in the document]
Comment (KanjiMonster) [reply recipes-kernel/linux/linux-yocto-onl_5.15.bb:23]: This is yocto-kernel specific due to the extra kernel-meta handling:

https://git.yoctoproject.org/poky/tree/meta/classes/kernel-yocto.bbclass?h=dunfell#n55

[the remark this answers is not in the document]
Comment (ideaship) [reply recipes-kernel/linux/linux-yocto-onl_5.15.bb:23]: Interesting. Thank you.

**Pull request merged**